mirascope/tools/base.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import ClassVar, Generic, TypeVar, cast
+
+from pydantic import BaseModel, ConfigDict, create_model
+
+from mirascope.core import BaseTool
+
+
+class _ToolConfig(BaseModel, ABC):
+    """Base configuration for tools"""
+
+    @classmethod
+    @abstractmethod
```

Comment:
seems like rather than making this abstract it should just return `cls()` by default? then tools that need to override this can, but those that don't then don't have to implement this method